Blog/views.py

```diff
@@ -146,12 +147,12 @@ class PostCreate(LoginRequiredMixin, generic.CreateView):
     def form_valid(self, form):
         form.instance.user = self.request.user
         form.instance.created = datetime.datetime.now()
-        same_titles_count = Post.objects.filter(title__iexact=form.instance.title).count()
-        slug = form.instance.title
-        if 0 < same_titles_count:
+        slug = slugify(form.instance.title)
+        same_slugs_count = Post.objects.filter(slug__icontains=form.instance.slug).count()
```

Comment:
You should filter for `slug` instead of  `form.instance.slug` because `form.instance.slug` is always an empty string at this point. Furthermore, this change should go onto the first commit, not onto the commit containing the user related code changes.